client/src/components/user/dashboard.js

```diff
@@ -72,10 +89,29 @@ class Dashboard extends React.Component {
     this.setState({ subscriptions });
   };
 
-  setEditingSubscription = async (editingSubscription) => {
-    await this.setState({ editingSubscription });
+  async handleDashboardChange(newView) {
+    const { mainComponentView } = this.state;
+
+    if (mainComponentView === 'dashboardCalendar' || mainComponentView === 'subscriptionDetail') {
+      return this.setState({ mainComponentView: newView });
+    }
+
+    const modal = new Modal(document.getElementById('myModal'));
+    modal.show();
+
+    const userInput = await waitUserClick();
+    if (userInput === 'next') {
+      await this.setState({ mainComponentView: newView });
+    } else if (userInput === 'close') {
+      await this.setState({ mainComponentView });
+    }
+    return modal.hide();
```

Comment:
I don't think you need to `return` here since this is just the end of the function

```suggestion
    modal.hide();
```

Reply:
return modal.hide is used because i have an if-statement that returns from the function early. ESLINT throws an error if you do not have a return statement at the end due to this.

Reply:
https://eslint.org/docs/rules/consistent-return